File: 프로그래머스/LV2/코딩테스트 연습/스택,큐/주식가격/주식가격.py
def solution(prices):
    answer = [i for i in range(len(prices)-1, -1, -1)]
    stack = []

    for i in range(len(prices)):
        while stack and stack[-1][1] > prices[i]:
            idx = stack.pop()[0]
            answer[idx] = i - idx

        stack.append([i, prices[i]])
    return answer

File: 프로그래머스/LV2/코딩테스트 연습/스택,큐/주식가격/주식가격.cpp
#include <string>
#include <vector>
#include <stack>
#define idx_price pair<int,int>
using namespace std;

vector<int> solution(vector<int> prices) 
{
    vector<int> answer;
    stack<idx_price> stack;
    int idx;
    for(int i = prices.size()-1; i >= 0; i--)answer.push_back(i);

    for(int i = 0; i < prices.size(); i++)
    {
        while (!stack.empty() && stack.top().second > prices[i])
        {
            idx = stack.top().first;
            answer[idx] = i - idx;
            stack.pop();
        }
        stack.push(idx_price(i, prices[i]));
    }
    return answer;
}